File: src/main/java/com/ism/dreamssellersv2/service/ItemService.java
package com.ism.dreamssellersv2.service;

import com.ism.model.ItemCreateDTO;
import com.ism.model.ItemDTO;
import com.ism.model.UpdateItemRequest;

import java.util.List;

public interface ItemService {
    List<ItemDTO> getItems();
    ItemDTO getItemById(Integer itemId);
    ItemDTO createItem(ItemCreateDTO itemCreateDTO);
    ItemDTO updateItem(Integer itemId, UpdateItemRequest updateItemRequest);
    void deleteItem(Integer itemId);
}

File: frontend/src/app/services/item.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ItemDTO, ItemCreateDTO } from '../models/item.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ItemService {
  private apiUrl = `${environment.apiUrl}/items`;

  constructor(private http: HttpClient) {}

  // Get all items
  getAllItems(): Observable<ItemDTO[]> {
    return this.http.get<ItemDTO[]>(this.apiUrl);
  }

  // Create new item
  createItem(item: ItemCreateDTO): Observable<ItemDTO> {
    return this.http.post<ItemDTO>(this.apiUrl, item);
  }

  // Get item by ID
  getItemById(itemId: number): Observable<ItemDTO> {
    return this.http.get<ItemDTO>(`${this.apiUrl}/${itemId}`);
  }

  // Delete item by ID
  deleteItem(itemId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${itemId}`);
  }
}
